File: src/dataset/c/isogram_exercism-pmarkaide/gemini/GeminiFlash_isogram_v4.c
#include "isogram.h"
#include <ctype.h>
#include <stdbool.h>

bool is_isogram(const char phrase[]) {
  if (phrase == NULL) {
    return false;
  }

  bool seen[26] = {false};
  for (int i = 0; phrase[i] != '\0'; ++i) {
    if (isalpha(phrase[i])) {
      int index = tolower(phrase[i]) - 'a';
      if (seen[index]) {
        return false;
      }
      seen[index] = true;
    }
  }
  return true;
}

File: src/dataset/javascript/isogram_exercism-elenamountz/gemini/GeminiFlash_isogram_v4.js
export const isIsogram = (str) => {
  const seen = new Set();
  for (let i = 0; i < str.length; i++) {
    const char = str[i].toLowerCase();
    if (/[a-z]/.test(char)) {
      if (seen.has(char)) {
        return false;
      }
      seen.add(char);
    }
  }
  return true;
};